### ISSUE TITLE
cli(suppression): unknown-metric-voids-marker contract untested through bca check

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#896]
## Summary

The CLI integration suite for suppression markers (`big-code-analysis-cli/tests/check_suppression.rs`) has no test that an **unknown metric identifier** inside `bca: suppress(...)` is rejected through the `bca check` binary — it must warn to stderr *and* void the entire marker so the violation still fires. Only the unknown-**verb** path (`// bca: allow(...)`, #263) is exercised at the integration level.

## Location

- `big-code-analysis-cli/tests/check_suppression.rs` (whole file; closest existing test is `legacy_allow_marker_does_not_suppress`, lines 159-189)

## Evidence

The "unknown identifier voids the entire marker" rule is a documented hard contract:

> spelling it `exit` is an unknown identifier, which warns *and voids the entire marker*
> — `big-code-analysis-book/src/commands/suppression.md` (Metric identifiers section)

The retired `exit` alias (#555) and any typo (`cyclomtic`) take the `SuppressionError::UnknownMetric` path, which renders a **distinct** warning from the verb path (`src/suppression.rs:233-249`):

```
unknown metric '{m}' in bca suppression marker; known metrics: abc, cognitive, ...
```

That warning reaches stderr via the same walk-time `eprintln!` the verb path uses (`src/spaces.rs:1102`), and the marker is dropped so the threshold fires.

The existing tests do **not** cover this through the binary:

- `legacy_allow_marker_does_not_suppress` (lines 159-189) pins the unknown-**verb** message (`unknown bca directive verb 'allow'`) — a different `SuppressionError` variant and a different rendered string.
- The library tests that do cover unknown-metric — `unknown_metric_in_marker_has_no_effect` (`tests/suppression_test.rs:257`) and `native_unknown_metric_errors` (`src/suppression.rs:666`) — assert only `FuncSpace.suppressed.is_empty()` / the error variant. Neither asserts that the binary emits the stderr warning and exits 2 with the violation visible.

So the CLI-surface behaviour (warn-to-stderr + fire-the-violation for an unknown *metric*) is unverified end-to-end.

## Expected Behavior

A `bca check` run over a fixture whose only suppression marker lists an unknown metric (e.g. `// bca: suppress(exit)` or a typo) should:

1. exit code 2 (violation reported — marker voided);
2. stderr names the offender and metric (`classify`, `cyclomatic`);
3. stderr carries the `unknown metric '...' in bca suppression marker` diagnostic.

This is the unknown-metric twin of `legacy_allow_marker_does_not_suppress`.

## Actual Behavior

No integration test exercises this path. A regression that (a) silently swallowed the unknown-metric warning, or (b) treated an unknown metric as a bare `suppress`-all (silencing the violation instead of voiding the marker), would pass the entire CLI suite — the most dangerous failure mode, since it would silently widen suppression scope on a typo.

## Impact

Authors and CI auditors rely on the void-on-typo contract: a misspelled metric must never silently silence a different (or every) metric. The library guards the data-structure half, but the user-visible CLI half (stderr diagnostic + exit code through `bca check`) is untested, so a regression there ships unnoticed.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

The CLI integration suite added a regression test for the unknown-**verb** path (`legacy_allow_marker_does_not_suppress`, #263) but never added the parallel unknown-**metric** path. The two take distinct `SuppressionError` variants and render distinct stderr strings, and the metric typo is the more common author error — yet only the verb half is pinned end-to-end through the binary.

### Implementation Steps

1. Add a fixture constant in `check_suppression.rs` mirroring `SUPPRESSED_RUST` but with an unknown metric in the list, e.g.:

   ```rust
   /// Rust function whose `bca: suppress(...)` lists an unknown metric.
   /// The retired `exit` alias (#555) is the realistic case: it must
   /// warn and void the *entire* marker, leaving the cyclomatic
   /// violation visible.
   const UNKNOWN_METRIC_RUST: &str = r#"
   pub fn classify(n: i32) -> &'static str {
       // bca: suppress(exit)
       if n < 0 { "neg" } else if n == 0 { "zero" } else { "pos" }
   }
   "#;
   ```

2. Add a test `unknown_metric_marker_does_not_suppress` that runs `check --threshold cyclomatic=1` and asserts, like the `allow` twin:
   - `.code(2)` — violation fires (marker voided, not silenced);
   - `stderr` contains `classify` and `cyclomatic`;
   - `stderr` contains `unknown metric 'exit' in bca suppression marker` (the distinct UnknownMetric diagnostic from `src/suppression.rs:247`).

   Pinning all three independently mirrors the rationale already documented in `legacy_allow_marker_does_not_suppress` (so a silent-swallow regression or a widen-to-suppress-all regression each surface clearly).

3. Optionally add a second arm asserting that a *valid* sibling metric in the same list is **not** honoured either (whole-marker void, not partial), e.g. `// bca: suppress(cyclomatic, exit)` still fires `cyclomatic=1` — this is the precise "voids the *entire* marker" wording from the docs.

### Tests to Add/Update

- New: `unknown_metric_marker_does_not_suppress` in `big-code-analysis-cli/tests/check_suppression.rs`.
- Optional: a `// bca: suppress(cyclomatic, exit)` whole-marker-void arm.

### Verification

- [ ] `cargo test -p big-code-analysis-cli` passes
- [ ] `cargo clippy -p big-code-analysis-cli` clean
- [ ] Test-via-revert: temporarily make the walker treat `UnknownMetric` as suppress-all (or swallow the warning) and confirm the new test fails on the exact assertion; restore.
- [ ] Manual: `bca check --paths <fixture-with-suppress(exit)> --threshold cyclomatic=1` exits 2 and prints the `unknown metric 'exit'` warning.

**Comment by dekobon** [comment]
Fixed on the working branch (commit `07dd9e61`).

Added `unknown_metric_voids_entire_marker` to `big-code-analysis-cli/tests/check_suppression.rs`. The fixture carries `// bca: suppress(cyclomatic, bogusmetric)` (one valid metric + one unknown) over a function that breaches `cyclomatic`, and the test pins three things independently through the `bca check` binary:

1. exit code 2 (the marker is voided, not silenced — guards against the dangerous suppress-all regression);
2. stderr names the offender (`classify`) and metric (`cyclomatic`);
3. stderr carries the distinct `unknown metric 'bogusmetric' in bca suppression marker` diagnostic.

Verified test-via-revert: perturbing `parse` in `src/suppression.rs:460-462` to skip the unknown metric instead of erroring made the run exit 0 with empty stderr (marker silenced the violation) and the test failed on exactly that. Production restored; test passes. The existing `suppression_marker_silences_violation_by_default` (a `suppress(cyclomatic)`-only marker that suppresses cleanly) serves as the valid-marker contrast, so no duplicate was added.

`cargo fmt --check`, `cargo clippy --workspace --all-targets -- -D warnings`, and `cargo test -p big-code-analysis-cli` all pass. Leaving open per instruction.
